docs/features/workflows.md

```diff
@@ -0,0 +1,53 @@
+# Workflows
+
+## Shopping Cart Event Workflow Events
+
+These events get triggered by `WorkflowShoppingCartEvents` which implements the `IShoppingCartEvents` interface. For each you can access the input as a .NET object using the `Context` workflow input and the serialized version as the `JSON` workflow input.
+All of these workflows expect to return one or more outputs which is passed back to the invoking code.
+
+> ⚠ If you want to return an altered version of the input as the output, please always use the JSON which is already serialized in the expected format used by OrchardCore.Commerce's converters. For example, you can use the JS expression `JSON.parse(input('JSON'))`.
+
+> ℹ When your output contains `LocalizedHtmlString`, it can be represented in JS either as `string` or `{ Name: string, Value: string }`. In case of just `string` the same text becomes `LocalizedHtmlString.Name` and `LocalizedHtmlString.Value` too.
+
+### "Cart displaying" Event
+
+Executes after the shopping cart data is prepared, but before the shapes are rendered.
+
+- Input: `ShoppingCartDisplayingEventContext` object containing the current shopping cart's headers and lines.
+- Outputs: either outputs are optional.
+  - Headers: `LocalizedHtmlString` array. The shopping cart header labels in order. If you have to support multiple locales, make sure to use the object format mentioned above, because `LocalizedHtmlString.Name` is used to generate the template name for the corresponding shopping cart column's cells.
+  - Lines: `ShoppingCartLineViewModel` array. This is only for display, in most cases, you shouldn't have to return this output.
+
+### "Verifying cart item" Event
+
+Executes before an item is added to the shopping cart to check whether it can be added based on inventory status.
+
+- Input: `ShoppingCartItem` object.
+- Outputs:
+  - Error: `LocalizedHtmlString` or `null`. The error message to be displayed if the input item can't be added to the cart. You can simply not output anything if the validation passes.
+
+### "Cart loaded" Event
```

Comment:
I'm only navigating to 1 product page, but it seems like the Cart Loaded event triggers 3 times always: 
![cart](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/80963259/a3d70bc1-fbbf-4cd7-baaf-8d28e347f7d1)

This is always the case with the Cart Loaded event.

Reply:
It's triggered each time the `IShoppingCartPersistence.RetrieveAsync()` is invoked so that's not surprising. Once by the shopping cart widget, but what's odd is that it's triggered twice by different method calls in `ShoppingCartHelpers.EstimateProductAsync()` used by the `DiscountPartDisplayDriver` to determine if any discounts are eligible. The latter probably can be avoided by optimization (so only load once during estimation and don't estimate if there is no discount to apply as is the case in your example).